Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncPerformanceContext.swift

```diff
@@ -0,0 +1,307 @@
+    var mongoClient: RemoteMongoClient?
+    var coll: RemoteMongoCollection<Document>?
+
+    var dbName = ""
+    var collName = ""
+    var userId = ""
+
+    let joiner = ThrowingCallbackJoiner()
+    let streamJoiner = StreamJoiner()
+    let testParams: TestParams?
+    let transport: FoundationInstrumentedHTTPTransport?
+    let harness: SyncPerformanceIntTestHarness?
+
+    init(harness: SyncPerformanceIntTestHarness,
+         testParams: TestParams,
+         transport: FoundationInstrumentedHTTPTransport) throws {
+        self.testParams = testParams
+        self.transport = transport
+        self.harness = harness
+        if testParams.stitchHostName == "https://stitch.mongodb.com" {
+            print("PerfLog: Context: Initializing STITCH-Prod")
+            dbName = harness.stitchTestDbName
+            collName = harness.stitchTestCollName
+            client = harness.outputClient
+            mongoClient = harness.outputMongoClient
+            coll = mongoClient?.db(dbName).collection(collName)
+            print("PerfLog: Context: Finished initializing STITCH-Prod")
+        } else {
+            print("PerfLog: Context: Initializing STITCH-Local")
+            dbName = ObjectId().oid
+            collName = ObjectId().oid
+
+            harness.setUp()
+            let app = try! harness.createApp()
+            _ = try! harness.addProvider(toApp: app.1, withConfig: ProviderConfigs.anon())
+            let svc = try! harness.addService(toApp: app.1, withType: "mongodb", withName: "mongodb1",
+                                         withConfig: ServiceConfigs.mongodb(name: "mongodb1", uri: harness.mongodbUri))
+            let rule = RuleCreator.mongoDb(database: dbName, collection: collName,
+                                           roles: [RuleCreator.Role(read: true, write: true)],
+                                           schema: RuleCreator.Schema())
+            _ = try! harness.addRule(toService: svc.1, withConfig: rule)
+
+            client = try! harness.appClient(forApp: app.0, withTransport: transport)
+            client?.auth.login(withCredential: AnonymousCredential(), joiner.capture())
+            let _: Any? = try joiner.value()
+            userId = client?.auth.currentUser?.id ?? ""
+
+            mongoClient = try! client?.serviceClient(fromFactory: remoteMongoClientFactory, withName: "mongodb1")
+            coll = mongoClient?.db(dbName).collection(collName)
+            print("PerfLog: Context: Finished Initializing STITCH-Local")
+        }
+
+        coll?.sync.proxy.dataSynchronizer.isSyncThreadEnabled = false
+        coll?.sync.proxy.dataSynchronizer.stop() // Failing here occaisonally
+        print("PerfLog: Context: Done with initialization")
+    }
+
+    func runSingleIteration(numDocs: Int,
+                            docSize: Int,
+                            testDefinition: TestDefinition,
+                            customSetup: TestDefinition? = nil,
```

Comment:
[nit] give different typealiases to these variables. They should also be named differently (`custom` is redundant), and instead of dealing with nullability, just make them empty closures.

Reply:
Kind of confused. Why give them different typealiases?  Also if I make them empty closures then they dont conform to the type alias which swift complains about

Reply:
Because even though the type definitions happen to be the same, they have semantically different uses.

To compile default args, you likely need to add empty params. So not:
`{}`
but:
`{ _, _, _ in }`
or something